src/cmd/windows/durable_task_monitor.go

```diff
@@ -0,0 +1,179 @@
+// +build windows
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+	"jenkinsci.org/plugins/durabletask/common"
+)
+
+type Shell int
+
+const (
+	CMD Shell = iota
+	POWERSHELL
+	PWSH
+)
+
+func (shell Shell) String() string {
+	switch shell {
+	case CMD:
+		return "cmd"
+	case POWERSHELL:
+		return "powershell"
+	case PWSH:
+		return "pwsh"
+	default:
+		return "UNKNOWN"
+	}
+}
+
+// Launches the script in a new session and waits for its completion.
+func launcher(wg *sync.WaitGroup, exitChan chan bool, shell string,
+	scriptPath string, resultPath string, outputPath string,
+	launchLogger *log.Logger, scriptLogger *log.Logger) {
+
+	defer wg.Done()
+	defer common.SignalFinished(exitChan)
+
+	if _, err := os.Stat(scriptPath); err != nil {
+		if os.IsNotExist(err) {
+			launchLogger.Printf("%s does not exist", scriptPath)
+			return
```

Comment:
should we not `Exit` with an errorcode here, or flag an error so that `main` can exit with a status code?

Reply:
resolved